### PR TITLE
Port bundle tile to C

### LINKED ISSUE #4649: Port bundle tile to C
- [x] HPACK
- [x] HTTP/2 framing layer
- [x] HTTP/2 client state machine
- [x] HTTP/2 stream demux
- [x] HTTP/2 header matcher
- [x] HTTP/2 non-blocking API syntax sugar
- [x] ~~HTTP/2 blocking API~~
- [x] gRPC receive buffer
- [x] gRPC user API
- [x] Bundle tile scaffold
- [x] Packet / bundle support
- [x] Keepalive
- [ ] Shredstream support 
- [x] Integration test against dummy bundle server
- [x] Remove Rust code
- [x] Production test (testnet)
- [x] Production test (mainnet)
- [x] DNS resolve on reconnect
- [ ] Metric for receive window issues

### PR DESCRIPTION
**Pull request by ripatel-fd**
Port bundle tile from Rust to C

Closes #4649

Depends on:
- https://github.com/firedancer-io/firedancer/pull/5170
- https://github.com/firedancer-io/firedancer/pull/5169
- https://github.com/firedancer-io/firedancer/pull/5168
- https://github.com/firedancer-io/firedancer/pull/5167
- https://github.com/firedancer-io/firedancer/pull/5194

### CODE REVIEW

**Comment by ripatel-fd** [comment]
@mmcgee-jump This is not feature-complete but ready to be merged:

Written and tested features:
- Packet and bundle streaming (modest 12 Gbps over loopback against the Rust dummy server)
- Conn- and stream-level flow control
- Reconnect with backoff
- HTTPS support via OpenSSL

Lacking tests:
- HTTP/2 framing layer edge cases (verified via h2spec but tests not yet codified)
- gRPC session layer (no unit tests)
- Bundle app layer (mostly lacking unit tests)

Not yet implemented:
- End-to-end tile level fuzzing
- Refresh DNS when reconnecting

File: src/disco/bundle/fd_bundle_tile.c [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+              fd_stem_context_t * stem,
+              int *               opt_poll_in,
+              int *               charge_busy ) {
+  (void)opt_poll_in;
+  if( FD_UNLIKELY( !ctx->stem ) ) ctx->stem = stem;
+  fd_bundle_client_step( ctx, charge_busy );
+
+  if( ctx->plugin_out.mem ) {
+    if( FD_UNLIKELY( ctx->bundle_status_recent != ctx->bundle_status_plugin ) ) {
+      fd_bundle_tile_publish_block_engine_update( ctx, stem );
```
Comment (mmcgee-jump) [inline src/disco/bundle/fd_bundle_tile.c:104]: *charge_busy = 1 here

File: src/disco/bundle/fd_bundle_auth.c [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    void const *         data,
+    ulong                data_sz
+) {
+  auther->needs_poll = 1;
+
+  pb_istream_t istream = pb_istream_from_buffer( data, data_sz );
+  auth_GenerateAuthChallengeResponse resp = auth_GenerateAuthChallengeResponse_init_default;
+  int decode_ok = pb_decode( &istream, &auth_GenerateAuthChallengeResponse_msg, &resp );
+  if( FD_UNLIKELY( !decode_ok ) ) {
+    FD_LOG_WARNING(( "Protobuf decode of (auth.GenerateAuthChallengeResponse) failed" ));
```
Comment (mmcgee-jump) [inline src/disco/bundle/fd_bundle_auth.c:79]: metrics over warnings plz

File: src/disco/bundle/fd_bundle_auth.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  char   challenge[ 9 ];
+  char   access_token [ 512 ];
+  ushort access_token_sz;
+};
+
+typedef struct fd_bundle_auther fd_bundle_auther_t;
+
+#define FD_BUNDLE_CLIENT_REQ_Auth_GenerateAuthChallenge         1
+#define FD_BUNDLE_CLIENT_REQ_Auth_GenerateAuthTokens            2
+//#define FD_BUNDLE_CLIENT_REQ_Auth_RefreshAccessToken            3
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_auth.h:21]: do we need this here?

File: src/disco/bundle/fd_bundle_auth.h [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#ifndef HEADER_fd_src_disco_bundle_fd_bundle_auth_h
+#define HEADER_fd_src_disco_bundle_fd_bundle_auth_h
+
+#include "../../waltz/grpc/fd_grpc_client.h"
+#include "../../disco/keyguard/fd_keyguard_client.h"
+
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_auth.h:6]: Do you think it's worth briefly outlining the expected auth flow at

File: src/disco/bundle/fd_bundle_client.c [hunk 5]
```diff
@@ -0,0 +1,898 @@
...
+void
+fd_bundle_tile_backoff( fd_bundle_tile_t * ctx,
+                        long               ts_ticks ) {
+  uint iter = ctx->backoff_iter;
+  if( ts_ticks < ctx->backoff_reset ) iter = 0U;
+  iter++;
+
+  /* FIXME proper backoff */
+  long wait_ticks = (long)( 500e6 * fd_tempo_tick_per_ns( NULL ) );
+  wait_ticks = (long)( fd_rng_ulong( ctx->rng ) & ( (1UL<<fd_ulong_find_msb_w_default( (ulong)wait_ticks, 0 ))-1UL ) );
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_client.c:343]: ignore if I misunderstood, but didn't you add the config option for this already? Why not use it?

File: src/disco/bundle/fd_bundle_client.c [hunk 6]
```diff
@@ -0,0 +1,898 @@
...
+  ulong tspub = (ulong)fd_frag_meta_ts_comp( fd_tickcount() );
+  fd_stem_publish( ctx->stem, ctx->verify_out.idx, sig, ctx->verify_out.chunk, sz, 0UL, 0UL, tspub );
+  ctx->verify_out.chunk = fd_dcache_compact_next( ctx->verify_out.chunk, sz, ctx->verify_out.chunk0, ctx->verify_out.wmark );
+  ctx->metrics.txn_received_cnt++;
+}
+
+/* Forwards a regular transaction to the tango message bus. */
+
+static void
+fd_bundle_tile_publish_txn(
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_client.c:412]: semi nit but maybe we can refactor these two publish_txn? majority of the code appears identical

File: src/disco/bundle/fd_bundle_client.c [hunk 7]
```diff
@@ -0,0 +1,898 @@
...
+  ulong tspub = (ulong)fd_frag_meta_ts_comp( fd_tickcount() );
+  fd_stem_publish( ctx->stem, ctx->verify_out.idx, sig, ctx->verify_out.chunk, sz, 0UL, 0UL, tspub );
+  ctx->verify_out.chunk = fd_dcache_compact_next( ctx->verify_out.chunk, sz, ctx->verify_out.chunk0, ctx->verify_out.wmark );
+  ctx->metrics.txn_received_cnt++;
+}
+
+/* Called for each transaction in a bundle.  Simply counts up
+   bundle_txn_cnt, but does not publish anything. */
+
+static bool
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_client.c:447]: Correct me if I'm wrong, but i thought we prefer to use int instead of bool?

File: src/disco/bundle/fd_bundle_client.c [hunk 8]
```diff
@@ -0,0 +1,898 @@
...
+  _Bool changed = !ctx->builder_info_avail;
+
+  block_engine_BlockBuilderFeeInfoResponse res = block_engine_BlockBuilderFeeInfoResponse_init_default;
+  if( FD_UNLIKELY( !pb_decode( istream, &block_engine_BlockBuilderFeeInfoResponse_msg, &res ) ) ) {
+    ctx->metrics.decode_fail_cnt++;
+    FD_LOG_WARNING(( "Protobuf decode of (block_engine.BlockBuilderFeeInfoResponse) failed" ));
+    return;
+  }
+  if( FD_UNLIKELY( res.commission > 100 ) ) {
+    FD_LOG_WARNING(( "BlockBuilderFeeInfoResponse commission out of range (0-100): %lu, assuming 100%% commission", res.commission ));
```
Comment (akhinvasara-jumptrading) [inline src/disco/bundle/fd_bundle_client.c:646]: Is this a valid assumption? What happens if wrong? 
and channeling michael, 'metrics?'  :)

File: src/disco/metrics/metrics.xml [hunk 9]
```diff
@@ -218,10 +218,24 @@ metric introduced.
     <counter name="PktRetransmissions" summary="Number of QUIC packets that retransmitted." />
 </tile>
 
+<enum name="BundleError">
```
Comment (mmcgee-jump) [inline src/disco/metrics/metrics.xml:221]: Can you add a success case here too? So the relative success rate can be calculated. It's unclear if BundleReceived is the same category, or if its counting after some other filtering

**Review by mmcgee-jump — APPROVED** [review]
